src/broker/saslAuthenticator/index.js

```diff
@@ -28,7 +28,12 @@ module.exports = class SASLAuthenticator {
   }
 
   async authenticate() {
-    const mechanism = this.connection.sasl.mechanism.toUpperCase()
+    let mechanism
+    if (typeof this.connection.sasl === 'function') {
+      mechanism = this.connection.sasl().mechanism.toUpperCase()
```

Comment:
Let's not duplicate the uppercasing. Instead, let's evaluate the function to get the sasl configuration if needed and then operate as normal:

```javascript
const saslConfiguration = typeof this.connection.sasl === 'function' ? this.connection.sasl() : this.connection.sasl
const mechanism = saslConfiguration.mechanism.toUpperCase()
```